From 06263de7bbf1bff94085d863b542e5005568d7dd Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:50 -0400
Subject: [ethernet] mlx4: Fix UP limit in ieee_ets->prio_tc

Message-id: <a12d2a9e487a90d9222b395888756cc06ebf3609.1410425016.git.avadai@redhat.com>
Patchwork-id: 93625
O-Subject: [Patch RHEL7.1 V1 054/197] net/mlx4_en: Fix UP limit in ieee_ets->prio_tc
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

User priority limit has to be less than MLX4_EN_NUM_UP.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Eugenia Emantayev <eugenia@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit fd8daa45f2bd9b876e0dbb9503ccc5a5252844f2)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_dcb_nl.c b/drivers/net/ethernet/mellanox/mlx4/en_dcb_nl.c
index b4881b6..c95ca25 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_dcb_nl.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_dcb_nl.c
@@ -62,7 +62,7 @@ static int mlx4_en_ets_validate(struct mlx4_en_priv *priv, struct ieee_ets *ets)
  int has_ets_tc = 0;
 
  for (i = 0; i < IEEE_8021QAZ_MAX_TCS; i++) {
-  if (ets->prio_tc[i] > MLX4_EN_NUM_UP) {
+  if (ets->prio_tc[i] >= MLX4_EN_NUM_UP) {
    en_err(priv, "Bad priority in UP <=> TC mapping. TC: %d, UP: %d\n",
      i, ets->prio_tc[i]);
    return -EINVAL;
-- 
1.7.1